docs/getting-started/cash-app-pay.md

```diff
@@ -0,0 +1,220 @@
+        is CashAppSignOrderResult.Success -> TODO("Create the PayKit customemr request")
+        is CashAppSignOrderResult.Failure -> TODO("Display an error and restart payment flow")
+      }
+    }
+  })
+}
+```
+
+### Step 5B: Sign the Order Token
+
+After retrieving the token from your server to server call, the order needs to be signed, so as to retrieve the JWT and associated data. This can be done either by the suspending function:
+
+``` kotlin
+Afterpay.signCashAppOrder(token)
+```
+
+or with the async version of it:
+
+``` kotlin
+Afterpay.signCashAppOrderAsync(token)
+```
+
+### Step 5C: Create a PayKit Customer Request
+
+To charge $5.00 as a one-time payment, your createrequest call might look like this (in the below example, `cashAppData` is the response object that is returned in the `didReceiveCashAppData` handler method):
+
+``` kotlin
+val request = PayKitPaymentAction.OneTimeAction(
+  redirectUri = cashAppData.redirectUri,
+  currency = PayKitCurrency.USD,
+  amount = (cashAppData.amount * 100).toInt(),
+  scopeId = cashAppData.merchantId,
+)
+
+payKit.createCustomerRequest(request)
+```
+
+## Step 6: Authorize the Customer Request
+
+### Step 6A: Add an Authorize Request Event to Cash App Pay button
+
+Once the Cash App PayKit SDK is in the `ReadyToAuthorize` state, you can display the Cash App Pay button. When the Customer taps the button, you can authorize the customer request. Documentation for a Cash App Pay button component can be found [here][cash-button-docs]{:target='_blank'}
+
+``` kotlin
+payKit.authorizeCustomerRequest()
+```
+
+{: .info }
+> Currently, the Button provided by the SDK is unmanaged. This means that it's a stylized button which isn't aware of SDK events out-of-the-box. It is the developer's responsibility to call the above method when the button is clicked and also manage any disabled and loading states.
+
+Your app will redirect to Cash App for authorization. When the authorization is completed, your redirect URI will be called to open your app. The SDK will fetch your authorized request and return it to your callback listener as one of 2 states: `Approved` or `Declined`.
+
+### Step 6B: Validate the Cash App Pay Order
+
+{: .alert }
+> This step must not be skipped
+
+Finally the Cash App order must be validated. This looks something like the below:
+
+``` kotlin
+Afterpay.validateCashAppOrder(
+    jwt,
+    grant.id,
+    customerResponseData.customerProfile!!.id,
+) { validationResult ->
+    when (validationResult) {
+      is CashAppValidationResponse.Success -> TODO("Capture payment with token and grant id")
+      is CashAppValidationResponse.Failure -> TODO("Handle an invalid Cash App order")
+    }
+}
+```
+
+## Step 7: Pass Grants to the Backend and Capture Payment
+
+The `Approved` state will contain a Grants list object associated with it and it can be used with Afterpay's capture payment API. Pass the grant id along with the token to capture using a server to server request.
```

Comment:
```suggestion
The `Approved` state will contain a Grants list object associated with it and it can be used with Afterpay's "Immediate Payment Capture" or "Deferred Payment Auth" API. Pass the grant ID along with the token to capture/auth using a server-to-server request.
```